### ISSUE TITLE
Whitelist a 100% GDPR compliant user experience company

### ISSUE THREAD

**Issue by bsafwen** [AdguardTeam/AdguardFilters#73968]
[//]: # (***You can delete or ignore strings starting with "[//]:" They will not be visible either way.)

***Description***:
* **Current behaviour**: Some GDPR compliant companies like contentsquare are blocked
Hello, thank you for the awesome job, I use ublockOrigin everyday and it would be nothing without your filters list so thank you.
I was just wondering if a company is 100% [GDPR](https://gdpr-info.eu/) compliant can be whitelisted.

Thanks.

**Comment by danidoedel** [comment]
GDPR compliant doesn‘t mean there are no trackers or no ads so I don‘t see the point in whitelisting? Also I don‘t think anyone here is a lawyer and able to check if it‘s really compliant :)

**Comment by gagarine** [comment]
A "good practices whitelist" is a very interesting idea. That could be an extra list users chose to activate (could be activated by default in the future).

But I believe GDPR compliant is not enough and also cover stuff not related to web page filtering. GDPR try to balance different interests, but Adguard is only about the end user, we can certainly make much simpler rules. Let's discuses the criteria.

For me I would definitely activate such a list if the website follow rules:
- No ads based on personal data
- Do not participate in ads network that engage in ads based on personal data
- No ads? (Nobody will activate it otherwise?)
- No annoyance (pop-up, cookie banners, excessive ads)
- No dark patterns
- No tracking of the user (opt-in tracking could be tolerated, I'm sure there is a case where it's OK)
- ...

Now this list could be super long. But also very quick to parse.

**Comment by bsafwen** [comment]
Yes, a whitelist of a good practices is very interesting.

For example, Contentsquare tracks user actions and helps e-commerce websites improve their UX, reise some elements, move them around, if a user is click raging on the checkout button contentsquare will raise an alert so that the e-commerce fix their checkout page, and a lot of stuff like this. The e-commerce website will ask the end user "Do you accept us collect anonmous data about your behaviour?", if the client says no then the contentsquare tag will not work, there is no ads, no popups nothing. All of this can be verified using the developer console in the network tab.

So as a user, it does not bother me at all to help my favorite e-commerce improve its UX, and of course I always have control by not giving my consent when I visit this website.

By blocking good practices companies, we favour big companies over smaller ones, because big companies can afford to internally implement the solution of tracking user actions, but smaller ones need a third party to do that for them.

**Comment by ameshkov** [comment]
Unblocking it in the tracking protection list would be against what users expect of us so we just cannot do that.

The problem with whitelists is the same as with ABP's ads whitelist - as long as there's just one party controlling it, there will always be questions to that list.

IMO, a good solution would be a more granular approach to lists - split it into more categories or even "tag" trackers - GDPR-compliant, transparency, static ads, etc, etc. However, reworking lists in such a way would be an extremely difficult task.

**Comment by gagarine** [comment]
I change my mind, I don't think it useful: If the service don't track their is no reason any of the resource will be blocked by ublock. If the user want to allow the service to track him, then he can disable ublock for this page. Message like "Do you accept us collect anonmous data about your behaviour?" is annoying. Life is too short to click on yes/no on every website.

What could be another ideas but totally out of the scope of Adguard is a way for website to signal trough the HTTP request or HTML they do not collect personal data, display ads based on personal data, etc. The reverse of https://globalprivacycontrol.org/ or "do-not-track".
